### PR TITLE
Streaming with pipes

### PR DESCRIPTION
**Pull request by fmthoma**

### CODE REVIEW

**Comment by quchen** [comment]
Is this a review request, or why did you assign it to me?

File: app/Main.hs [hunk 1]
```diff
@@ -32,57 +42,114 @@ main :: IO ()
...
 
-app :: Text -> App MainWidget
-app grepOutput = App
-    { _initialize  = initSplitView
-    , _handleEvent = eventHandler
-    , _render      = fmap picForImage . drawWidget }
+data AppState = AppState { _appWidget  :: MainWidget
+                         , _inputLines :: Seq Text }
+
+appWidget :: Lens' AppState MainWidget
```
Comment (quchen) [inline app/Main.hs]: Why not `makeLenses`?

File: app/Main.hs [hunk 2]
```diff
@@ -32,57 +42,114 @@ main :: IO ()
...
+appWidget = lens _appWidget (\s w -> s { _appWidget = w })
+
+inputLines :: Lens' AppState (Seq Text)
+inputLines = lens _inputLines (\s l -> s { _inputLines = l })
+
+data Event = VtyEvent Vty.Event
+           | ReceiveInputEvent  Text
+           | ReceiveResultEvent Text
+
+vtyEvent :: Prism' Event Vty.Event
```
Comment (quchen) [inline app/Main.hs]: Why not `makePrisms`?

Comment (fmthoma) [inline app/Main.hs]: Good question. Wanted to see the mechanics behind `makePrisms` and `makeLenses`, and it didn't hurt enough to change it afterwards.

**Pull request merged**